editoast/src/views/light_rolling_stocks.rs

```diff
@@ -0,0 +1,82 @@
+use crate::error::Result;
+use crate::schema::rolling_stock::LightRollingStock;
+use crate::DbPool;
+use actix_web::dev::HttpServiceFactory;
+use actix_web::get;
+use actix_web::web::{self, Data, Json, Path};
+
+pub fn routes() -> impl HttpServiceFactory {
+    web::scope("/light_rolling_stock").service((get, list))
+}
+
+#[get("")]
+async fn list(db_pool: Data<DbPool>) -> Result<Json<Vec<LightRollingStock>>> {
+    match LightRollingStock::list(db_pool).await {
+        Ok(rolling_stocks) => Ok(Json(rolling_stocks)),
+        Err(e) => Err(e),
+    }
+}
```

Comment:
Two things here. 

- We should return a paginated response.
- We can use `?` to avoid matching the response.

```suggestion
#[get("")]
async fn list(
    db_pool: Data<DbPool>,
    page_settings: Query<PaginationQueryParam>,
) -> Result<Json<PaginatedResponse<LightRollingStock>>> {
    let page = page_settings.page;
    let per_page = page_settings.page_size.unwrap_or(25);
    Ok(Json(
        LightRollingStock::list(db_pool, page, per_page).await?,
    ))
}
```

Reply:
Ok thanks, I thought the page should not be paginated (as in the front, we query all the rolling_stocks at once), but I can fix this :+1: